src/chain/chain.rs

```diff
@@ -41,10 +41,19 @@ pub fn delivery_group_size(n: usize) -> usize {
     (n + 2) / 3
 }
 
+/// Network parameters: number of elders, safe section size
+#[derive(Clone, Copy, Debug)]
+pub struct NetworkParams {
+    /// The number of elders per section
+    pub elder_size: usize,
+    /// Minimum number of nodes we consider safe in a section
+    pub safe_section_size: usize,
+}
+
 /// Data chain.
 pub struct Chain {
-    /// Minimum number of nodes we consider acceptable in a section
-    min_sec_size: usize,
+    /// Network parameters
+    network_cfg: Option<NetworkParams>,
```

Comment:
Minor suggestion: what about changing this to just `network_cfg: NetworkParams` and defining `Default` impl for `NetworkParams`. Then pass `NetworkParam::default()` everywhere you pass `None` now. I think it could make the code a bit simpler and also allow to override just one of the params while keeping the other one default:
```rust
NetworkParams {
    safe_section_size: blah,
    ..Default::default()
}
```
This comment applies to all the other places where `Option<NetworkParams>` is used currently.

Reply:
That makes a lot of sense. @dirvine would you be OK with me implementing this solution?

Reply:
I would indeed.